### ISSUE TITLE
Tasks do not emit errors in the same way their command-line equivalents do, causing unexpected behaviour

### ISSUE THREAD

**Issue by IanKemp** [microsoft/azure-pipelines-tasks#11238]
**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: NuGetCommand@2, PublishBuildArtifacts@1, many others

## Environment
- Server: Azure Pipelines
- Agent: Hosted

## Issue Description

My company has a project, originally in .NET Framework, that was built into a NuGet package via a build pipeline in Azure DevOps. The package was created via a nuspec file and the `nuget pack` task:

![image](https://user-images.githubusercontent.com/1726434/63840956-d1d2f500-c979-11e9-93e6-c4dac45dd250.png)

This project was migrated to .NET Standard but the nuspec was retained. I then deleted the nuspec file (as that package metadata is now built into the .NET Standard/Core csproj file format, making the nuspec redundant) but neglected to update the task to look at the csproj instead of the nuspec.

The result is that when the build is run, the `nuget pack` task looks for a nonexistent nuspec file, and of course doesn't find it. But... it doesn't have any problem with that fact. It doesn't error, it doesn't even warn, it just completes successfully. I had to set `system.debug = true` to even get any inkling of a problem ("Found 0 files", [see attached log file](https://github.com/microsoft/azure-pipelines-tasks/files/3550092/task.log)).

The end result is that no NuGet package is created, which means that the subsequent "Publish artifacts" task does nothing. Which means our release pipeline's `nuget push` does nothing. Which means that no NuGet package is uploaded to our feed. Which makes my clients unhappy. Which makes me very unhappy.

Did I mention that there are no errors at any stage of this process? The artifact publish and NuGet push at least give warnings - but warnings don't fail a pipeline, nor can they be made to.

Even better, if I do the following on my local machine:

```
> nuget pack <project-name>/<project-name>.nuspec
```

I get:

```
Attempting to build package from '<project-name>.nuspec'.
Error NU5008: Manifest file not found at 'C:\Projects\<project-name>\<project-name>.nuspec'
```

In other words, nuget.exe fails hard when it can't find the specific file it's asked to pack with, but the Azure Pipelines NuGet task doesn't. Why is the behaviour different?

More to the point, why are you guys so apparently averse to emitting errors from your tasks, especially when the "Continue on error" checkbox exists? Warnings are not acceptable, because a file that is not found almost certainly means that there is a problem with the pipeline configuration, and I want to know about a problem as soon as it happens, not have to go hunting through logs to find it after the fact.

It isn't Azure's place to decide whether a missing file is fine or not, it's mine. And currently I have no option to make that decision. So you need to add it.

## Fix

1. The `nuget pack` Azure Pipelines task must behave identically to its command-line counterpart when confronted with a path that doesn't exist. In other words, instead of doing nothing, it should give a very explicit error, then terminate.
2. All tasks that deal with file paths must have a boolean option (default false) to fail immediately if the provided file path or glob expression yields no matching files.
3. All tasks must have a boolean option (default false) to treat warnings as errors.

**Comment by IanKemp** [comment]
This issue is not stale, the team simply hasn't bother to address it. Presumably they hope it will age away so they don't have to do actual work.

**Comment by sim590** [comment]
Has anything been done for this? We have a pipeline which is trying to push a nuget package file, but we made a mistake in naming the file, but the pipeline didn't report an error and the package was not pushed. We had to find out that there was a WARNING responsible for this:

![Image](https://github.com/user-attachments/assets/89a4ef0a-ac5e-4550-b153-b7c55c709d29)

But this should have been an ERROR and not a WARNING.
